lib/utils/error.js

```diff
@@ -1,9 +1,16 @@
-class SpeechCredentialError extends Error {
+class NonFatalTaskError extends Error {
```

Comment:
I think I would prefer to have a subclass of NonFatalTaskError, e.g

```
class PlayUrlNotFound extends NonFatalTaskError {
  constructor(url) {
    super('File not found');
    this.url = url; 
  }
```

Then in play.js throw that `PlayUrlNotFound`.  That way in our log messages we would get the specific URL not found I think, instead of this

```
{"level":50,"time":1733165080947,"pid":65211,"hostname":"daveh-macbook-pro.local","callId":"1-1@172.38.0.10-31","callSid":"7135de45-436f-450a-a3ed-1d379e9a6e10","accountSid":"bb845d4b-83a9-4cde-a6e9-50f3743bab3f","callingNumber":"play_single_link","calledNumber":"16174000000","traceId":"00000000000000000000000000000000","err":{"type":"NonFatalTaskError","message":"File not Found","stack":"Error: File not Found\n    at TaskPlay.exec (/Users/dhorton/tmp/jambonz-feature-server/lib/tasks/play.js:75:13)\n    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)\n    at async InboundCallSession.exec (/Users/dhorton/tmp/jambonz-feature-server/lib/session/call-session.js:1084:11)"},"msg":"Error executing task  #0:1: play"}
```

Reply:
in CallSession we can still test `instanceOf NonFatalTaskError` though because that will return true when testing if B is an instanceOf A when B subclasses A